modules/siddhi-extensions/input-mappers/json-input-mapper/src/main/java/org/wso2/siddhi/extension/input/mapper/json/JsonSourceMapper.java

```diff
@@ -0,0 +1,498 @@
+                        description =  "Above configuration will do a default JSON input mapping. Expected "
+                                + "input will look like below."
+                                + "{\n"
+                                + "    \"event\":{\n"
+                                + "        \"symbol\":\"WSO2\",\n"
+                                + "        \"price\":55.6,\n"
+                                + "        \"volume\":100\n"
+                                + "    }\n"
+                                + "}\n"),
+                @Example(
+                        syntax = "@source(type='inMemory', topic='stock', @map(type='json', "
+                                + "enclosing.element=\"$.portfolio\", "
+                                + "@attributes(symbol = \"company.symbol\", price = \"price\", volume = \"volume\")))",
+                        description =  "Above configuration will perform a custom JSON mapping. Expected input will "
+                                + "look like below."
+                                + "{"
+                                + " \"portfolio\":{\n"
+                                + "     \"stock\":{"
+                                + "        \"volume\":100,\n"
+                                + "        \"company\":{\n"
+                                + "           \"symbol\":\"WSO2\"\n"
+                                + "       },\n"
+                                + "        \"price\":55.6\n"
+                                + "    }\n"
+                                + "}")
+        }
+)
+
+public class JsonSourceMapper extends SourceMapper {
+
+    private static final String DEFAULT_JSON_MAPPING_PREFIX = "$.";
+    private static final String DEFAULT_JSON_EVENT_IDENTIFIER = "event";
+    private static final String DEFAULT_ENCLOSING_ELEMENT = "$";
+    private static final String FAIL_ON_MISSING_ATTRIBUTE_IDENTIFIER = "fail.on.missing.attribute";
+    private static final String ENCLOSING_ELEMENT_IDENTIFIER = "enclosing.element";
+    private static final Logger log = Logger.getLogger(JsonSourceMapper.class);
+
+    private StreamDefinition streamDefinition;
+    private MappingPositionData[] mappingPositions;
+    private List<Attribute> streamAttributes;
+    private boolean isCustomMappingEnabled = false;
+    private boolean failOnMissingAttribute = true;
+    private String enclosingElement = null;
+    private AttributeConverter attributeConverter = new AttributeConverter();
+    private JsonFactory factory;
+    private int attributesSize;
+
+    @Override
+    public void init(StreamDefinition streamDefinition, OptionHolder optionHolder,
+                     List<AttributeMapping> attributeMappingList, ConfigReader configReader) {
+        this.streamDefinition = streamDefinition;
+        this.streamAttributes = this.streamDefinition.getAttributeList();
+        attributesSize = this.streamDefinition.getAttributeList().size();
```

Comment:
Use this. and keep consistency across.

Reply:
Fixed with PR https://github.com/wso2-extensions/siddhi-map-json/pull/56